File: src/main/java/ipp/IppVersionsSupported.java
package ipp;

import de.gmuth.ipp.client.IppPrinter;
import java.util.List;

public class IppVersionsSupported {

  public static void main(String[] args) {
    IppPrinter ippPrinter = new IppPrinter("ipp://localhost:8632/printers/laser");
    List<String> ippVersionsSupported = ippPrinter.getAttributes().getValues("ipp-versions-supported");
    System.out.println("ipp versions supported: " + ippVersionsSupported);
  }

  // ipp versions supported: [1.0, 1.1, 2.0]

}

File: src/main/kotlin/ipp/IppVersionsSupported.kt
package ipp

import de.gmuth.ipp.client.IppPrinter
import ipp.Printers.laser

fun main() {
    val ippPrinter = IppPrinter(laser)
    val ippVersionsSupported: List<String> = ippPrinter.attributes.getValues("ipp-versions-supported")
    println("ipp versions supported: $ippVersionsSupported")

    // ipp versions supported: [1.0, 1.1, 2.0]
}